src/meta/uchange.lean

```diff
@@ -0,0 +1,68 @@
+/-
+Copyright (c) 2020 Gabriel Ebner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gabriel Ebner
+-/
+
+/-!
+# Changing universes of types in meta-code
+
+This file defines the meta type `uchange (α : Type v) : Type u`, which
+permits us to change the universe of a type analogously to `ulift`.
+However since `uchange` is meta, it can both lift and lower the universe.
+
+The implementation of `uchange` is efficient. Both `uchange.up` and
+`uchange.down` compile to no-ops.
+-/
+
+universes u v
+
+/--
+`unchecked_cast' a : β` performs an unchecked cast of `(a : α)` to `β`.
+
+Unlike `unchecked_cast`, it can cast across universes. The VM implementation
+is guaranteed to be the identity.
+-/
+@[inline, irreducible]
+meta def unchecked_cast' {α : Sort u} {β : Sort v} (a : α) : β :=
+plift.down $ @cast (α → β → plift β) (β → α → plift β) undefined (λ _ a, plift.up a)
+  (cast undefined punit.star) a
+
+/--
+`uchange (α : Sort v) : Sort u` is an equivalent type in a different universe.
+
+In the VM, both `α` and `uchange α` have the same representation.
```

Comment:
```suggestion
In the VM, both `α` and `uchange α` have the same representation.

This definition is `meta` because it collapses the universe hierarchy; if pure code could do
this then one could derive Girard's paradox.
```